src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs

```diff
@@ -631,5 +648,133 @@ static string GetJsonText(string text)
 
             return result;
         }
+
+        private void WriteWarning(string message)
+        {
+            EventHandler<StreamEventArgs> writeWarningEvent;
+            if (AzureSession.Instance.TryGetComponent(AzureRMCmdlet.WriteWarningKey, out writeWarningEvent))
+            {
+                writeWarningEvent(this, new StreamEventArgs() { Message = message });
```

Comment:
This is fine, but I think the implementation on the other end may lead to issues.